dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ResolveDirector.java

```diff
@@ -641,8 +641,24 @@ void resolveAutoFeatures(Collection<ESAAsset> autoFeatures, ArrayList<InstallAss
                 Set<String> allServerFeatures = new HashSet<>(InstallUtils.getAllServerFeatures());
                 allServerFeatures.addAll(assetsToInstall);
 
-                log(Level.FINE, "Using resolveAsSet to resolve features");
+                // FIRST CALL to resolveAsSet --> this throws an error for singleton features
+                log(Level.FINE, "Calling resolveAsSet api");
                 installResources = resolver.resolveAsSet(allServerFeatures); // use new api
+
+
+                Set<String> returnedFeatureList = new HashSet<>();
+                for(List<RepositoryResource>  resList  : installResources){
+                    for(RepositoryResource res : resList) {
+                        if (res.getType().equals(ResourceType.FEATURE)) {
+                            EsaResource esa = (EsaResource) res;
+                            returnedFeatureList.add(esa.getProvideFeature());
+                        }
+                    }
+                }
+                InstallUtils.setResolveAsSetFeatures(returnedFeatureList);
```

Comment:
You should skip the 2nd call to resolve the auto features if returnedFeatureList is empty.

Reply:
fixed in commit https://github.com/OpenLiberty/open-liberty/pull/11203/commits/54588424ec43750b0f0e27d631b3155647c611a4